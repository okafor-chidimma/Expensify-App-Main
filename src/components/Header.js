import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOutProcess } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>

    <NavLink to='/dashboard' activeClassName='is-active'>
      My Dashboard
    </NavLink>
    <NavLink to='/create' activeClassName='is-active'>
      Create Expense
    </NavLink>
    {
      // <NavLink to="/edit/:id" activeClassName="is-active">Edit</NavLink>
      // <NavLink to='/help' activeClassName='is-active'>
      //   Help
      // </NavLink>
    }
    <button onClick={startLogout}>Log out</button>
  </header>
);

const matchDispatchToState = dispatch => ({
  startLogout: () => dispatch(startLogOutProcess())
});
export default connect(undefined, matchDispatchToState)(Header);
