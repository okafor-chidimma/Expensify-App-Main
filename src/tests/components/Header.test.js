import React from 'react';
import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import Header from '../../components/Header';
describe('To test the Header component with enzyme package', () => {
  test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('h1').text()).toBe('Expensify');
    expect(wrapper).toMatchSnapshot();
    // expect(toJson(wrapper)).toMatchSnapshot();
    // console.log(wrapper.debug());
    // console.log(wrapper);
    // console.log(toJson(wrapper));
    // console.log(<Header />);

  })
})


