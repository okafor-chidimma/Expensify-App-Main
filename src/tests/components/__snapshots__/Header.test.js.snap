// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the Header component with enzyme package should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    to="/dashboard"
  >
    My Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    LogOut
  </button>
</header>
`;
